lib/stanford-mods/date_parsing.rb

```diff
@@ -8,10 +8,59 @@ module Mods
     #       or have an initialize method so we don't have to keep passing the date_str argument
     class DateParsing
 
-      # looks for dddd pattern in String and returns it if found
+      # get single facet value for date, generally an explicit year or "17th century" or "5 B.C."
+      # @param [String] date_str String containing a date (we hope)
+      # @return [String, nil] String facet value for year if we could parse one, nil otherwise
+      def self.facet_string_from_date_str(date_str)
+        # B.C. first in case there are 4 digits, e.g. 1600 B.C.
+        return facet_string_for_bc(date_str) if date_str.match(BC_REGEX)
+        # most date strings have a four digit year
+        result = sortable_year_for_yyyy(date_str)
+        # 2 digit year will always be 19xx or 20xx; sortable version will make a good facet string
+        result = sortable_year_for_yy(date_str) unless result
+        # decades are always 19xx or 20xx; sortable version will make a good facet string
+        result = sortable_year_for_decade(date_str) unless result
+        result = facet_string_for_century(date_str) unless result
+        result = facet_string_for_early_numeric(date_str) unless result
+        unless result
+          # try removing brackets between digits in case we have 169[5] or [18]91
+          if date_str.match(/\d[\[\]]\d/)
+            no_brackets = date_str.delete('[').delete(']')
+            result = facet_string_from_date_str(no_brackets)
+          end
+        end
+        result
+      end
+
+      # get String sortable value year if we can parse date_str to get a year.
+      #   SearchWorks currently uses a string field for pub date sorting; thus so does Spotlight.
+      #   The values returned must *lexically* sort in chronological order, so the B.C. dates are tricky
+      # @param [String] date_str String containing a date (we hope)
+      # @return [String, nil] String sortable year if we could parse one, nil otherwise
+      #  note that these values must *lexically* sort to create a chronological sort.
+      def self.sortable_year_string_from_date_str(date_str)
+        # B.C. first in case there are 4 digits, e.g. 1600 B.C.
+        return sortable_year_for_bc(date_str) if date_str.match(BC_REGEX)
+        # most date strings have a four digit year
+        result = sortable_year_for_yyyy(date_str)
+        result = sortable_year_for_yy(date_str) unless result
+        result = sortable_year_for_decade(date_str) unless result
+        result = sortable_year_for_century(date_str) unless result
+        result = sortable_year_for_early_numeric(date_str) unless result
+        unless result
+          # try removing brackets between digits in case we have 169[5] or [18]91
+          if date_str.match(/\d[\[\]]\d/)
+            no_brackets = date_str.delete('[').delete(']')
```

Comment:
ah - when I looked at the documentation, I thought it would delete the arg as a string match, not the individual chars.